src/dht.cpp

```diff
@@ -2052,10 +2056,27 @@ Dht::onDisconnected()
     }
     if (bootstrapJob)
         bootstrapJob->cancel();
-    bootstrapJob = scheduler.add(scheduler.time() + bootstrap_period, std::bind(&Dht::onDisconnected, this));
+    bootstrapJob = scheduler.add(scheduler.time() + bootstrap_period, std::bind(&Dht::bootstrap, this));
```

Comment:
I still not really understand how it can cause https://github.com/savoirfairelinux/opendht/issues/575

I mean, the bootstrap_period should never be < 10 secs.

The only thing I see, ::time() may return an old value causing now + 10 to be < to the actual now() (and causing an infinite loop), maybe a missing syncTime() ? Maybe add syncTime in ::bootstrap() ?

Reply:
Main issue is fixed by this commit. Overflow after 252 years to be handled in another commit.

Reply:
done